### PR TITLE
Fully remove pendulum from dagster core

### PR DESCRIPTION
**Pull request by gibsondan**
## Summary & Motivation
Will wait to land this until after the 1.8 branch cut.

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/time_window_partitions.py [hunk 1]
```diff
@@ -128,27 +120,20 @@ def dst_safe_strptime(date_string: str, tz: str, fmt: str) -> PendulumDateTime:
     # the datetime object may have timezone information on it, depending on the format used. If it
     # does, we simply ensure that this timestamp is in the correct timezone.
     if dt.tzinfo:
-        return pendulum.from_timestamp(dt.timestamp(), tz=tz)
+        return datetime.fromtimestamp(dt.timestamp(), tz=get_timezone(tz))
     # otherwise, ensure that we assume the pre-transition timezone
     else:
-        # Pendulum 1.x erroneously believes that there are two instances of the *second* hour after
-        # a datetime transition, so to work around this we calculate the timestamp of the next
-        # microsecond of the given datetime.
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/definitions/time_window_partitions.py:136]: won't miss this!

File: .buildkite/dagster-buildkite/dagster_buildkite/steps/packages.py [hunk 2]
```diff
@@ -360,10 +360,6 @@ def _unsupported_dagster_python_versions(tox_factor: Optional[str]) -> List[Avai
         return [AvailablePythonVersion.V3_11, AvailablePythonVersion.V3_12]
 
     if tox_factor in {
-        "definitions_tests_pendulum_1",
```
Comment (OwenKephart) [inline .buildkite/dagster-buildkite/dagster_buildkite/steps/packages.py:363]: glorious

File: python_modules/dagster/dagster/_seven/compat/pendulum.py [hunk 3]
```diff
@@ -1,132 +0,0 @@
-import datetime
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_seven/compat/pendulum.py:1]: wowee

**Review by OwenKephart — APPROVED** [review]
amazing!

**Pull request merged**